### ISSUE TITLE
[Feature Request] GlSL macro

### ISSUE THREAD

**Issue by tmvkrpxl0** [glium/glium#2050]
yew has html macro like this:
![image](https://user-images.githubusercontent.com/41675181/227732787-3b42e769-cddb-40f5-9d60-212813e28372.png)
Which allows rust compiler to verify syntax of html code and even highlight errors.
I admit that it would be hard to implement such thing in glium, but still, it would greatly simplify implementing shader, as not only devs can work on it in rust file, but it can also be extended to treat uniforms also as variable reference
```rs
let t = 0.0; // this will be uniform
let shader = program! {
  vertex! {
    in vec2 position;
    
    uniform float t;
    
    void main() {
        vec2 pos = position;
        pos.x += t;
        gl_Position = vec4(pos, 0.0, 1.0);
    }
  }
}
```
If variable t in this case is i32, it would cause compiler error

**Comment by tmvkrpxl0** [comment]
Sorry, I should've researched more, I've just discovered that there's macro called program already.
Still, expanding its capability to such level would be nice addition

**Comment by Melchizedek6809** [comment]
Yeah, it does sounds a bit too ambitious for glium. A big feature like this should also probably go in a separate crate to keep glium somewhat small.

**Comment by tmvkrpxl0** [comment]
https://github.com/vtavernier/glsl-lang and https://github.com/phaazon/glsl
For future record, thees already implemented macro for it
